devops/azure-pipeline.yml

```
@@ -0,0 +1,211 @@
+        displayName: Quality Analysis
+        workspace:
+          clean: all
+        steps:
+          - task: GoTool@0
+            displayName: Use Go 1.13.14
+            inputs:
+              version: 1.13.14
+          - task: Go@0
+            displayName: go mod download
+            inputs:
+              command: custom
+              customCommand: mod
+              arguments: download
+              workingDirectory: $(System.DefaultWorkingDirectory)
+          - task: PowerShell@2
+            condition: eq(variables['LintExecute'], 'true')
+            displayName: Install Go lint
+            inputs:
+              targetType: inline
+              script: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(System.DefaultWorkingDirectory) v1.42.1
+          - task: PowerShell@2
+            condition: eq(variables['LintExecute'], 'true')
+            displayName: Run linter
+            inputs:
+              targetType: inline
+              script: $(System.DefaultWorkingDirectory)/golangci-lint run --new-from-rev=origin/master --timeout 2m0s --allow-parallel-runners
+          - task: Go@0
+            displayName: go test
+            inputs:
+              command: test
+              arguments: ./... -short -coverprofile=coverage.file -tags=unit
+              workingDirectory: $(System.DefaultWorkingDirectory)
+
+          - task: SonarCloudPrepare@1
+            displayName: Prepare analysis on SonarCloud
+            inputs:
+              SonarCloud: $(SonarcloudAccount)
+              organization: $(Organization)
+              projectKey: $(ProjectKey)
+              projectName: $(ProjectName)
+              scannerMode: CLI
+              configFile: $(System.DefaultWorkingDirectory)/sonar-project.properties
+          - task: SonarCloudAnalyze@1
+            displayName: Run Code Analysis
+          - task: SonarCloudPublish@1
+            displayName: Publish Quality Gate Result
+      - job: ContractTest
+        displayName: Contract Test
+        dependsOn: QualityAnalysis
+        condition: |
+          and
+          (
+            eq(variables['ContractTestExecute'], 'true'),
+            eq(dependencies.QualityAnalysis.result, 'Succeeded')
+          )
+        steps:
+          - task: Bash@3
+            displayName: docker-compose build contract-test
+            inputs:
+              targetType: inline
+              script: >
+                cd devops
+
+                docker-compose build contract-test
+          - task: Bash@3
+            displayName: Run contract test
+            inputs:
+              targetType: inline
+              script: >-
+                cd devops
+
+                docker-compose run --rm -e PACT_BROKER_TOKEN=$(PACT_BROKER_TOKEN) -e PACT_URL=$(PACT_URL) -e GITHUB_COMMIT=$(Build.SourceVersion) -e GITHUB_BRANCH=$(Build.SourceBranchName) -e PACT_BROKER_URL=https://pagarme.pactflow.io/ contract-test

```

O que acha de quebrar o comando em várias linhas para facilitar a leitura?
```suggestion
                docker-compose run --rm \
                -e PACT_BROKER_TOKEN=$(PACT_BROKER_TOKEN) \
                -e PACT_URL=$(PACT_URL) \
                -e GITHUB_COMMIT=$(Build.SourceVersion) \
                -e GITHUB_BRANCH=$(Build.SourceBranchName) \
                -e PACT_BROKER_URL=https://pagarme.pactflow.io/ \
                contract-test
```

---

Uma dúvida breve não relacionado à pipeline:
O `PACT_BROKER_URL` do teste de contrato ficou como variável de ambiente?

---

@PauloGoncalvesBH ficou sim! aqui https://github.com/mundipagg/boleto-api/blob/db4e7159b1cc86b4919bcd499863861ae2c57186/test/boleto_pact_provider_test.go#L137